apps/total-typescript/src/hooks/use-survey.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import {QuestionProps} from '@skillrecordings/quiz/dist/components/question'
+import {QuizContext} from '@skillrecordings/quiz/dist/machines/quiz-machine'
+import {surveyData, surveyConfig} from 'components/survey/survey-config'
+import {QuestionResource} from '@skillrecordings/types'
+import {useQuestion} from '@skillrecordings/quiz'
+import {useConvertkit} from './use-convertkit'
+import {isBefore, subDays} from 'date-fns'
+import {useReward} from 'react-rewards'
+import {useRouter} from 'next/router'
+import {SURVEY_ID} from 'pages/ask'
+import {trpc} from '../utils/trpc'
+import isEmpty from 'lodash/isEmpty'
+import get from 'lodash/get'
+
+type SurveyContextType = {
+  question: QuestionProps
+  popup: {
+    isPopupClosed: boolean
+    handleClose: () => void
+    handleDontSurvey: () => void
+  }
+}
+
+export const SurveyContext = React.createContext({} as SurveyContextType)
+
+type SurveyProviderProps = {
+  currentQuestion?: QuestionResource
+}
+
+const DAYS_TO_WAIT_BETWEEN_QUESTIONS = 3
+
+export const SurveyProvider: React.FC<
+  React.PropsWithChildren<SurveyProviderProps>
+> = ({children}) => {
+  const {subscriber, loadingSubscriber} = useConvertkit()
+  const currentSurveySlug = SURVEY_ID
+  const survey = get(surveyData, currentSurveySlug)
+  const answerSurveyMutation = trpc.useMutation(['convertkit.answerSurvey'])
+  const [currentQuestion, setCurrentQuestion] = React.useState<string>('')
+
+  const question = useQuestion({
+    currentQuestion: survey.questions[currentQuestion],
+    currentQuestionKey: currentQuestion,
+    questionSet: survey.questions,
+    config: surveyConfig,
+    handleSubmitAnswer: async ({
+      answer,
+      currentQuestionKey: question,
+    }: QuizContext) => {
+      if (answer && question) {
+        answerSurveyMutation.mutate({
+          answer,
+          question,
+        })
+      }
+    },
+  })
+
+  React.useEffect(() => {
+    if (subscriber && subscriber.fields) {
```

Comment:
@joelhooks while testing this I noticed that I often ended up not having anything but an `id` in my `ck_subscriber` cookie (no `fields`, etc). would it make sense to call `updateSubscriber` and populate cookie with fresh data more often? e.g. when completing a lesson or subscribing.